File: Ahorn/entities/CoreModeMusicController.jl
module CommunalHelperCoreModeMusicController

using ..Ahorn, Maple

@mapdef Entity "CommunalHelper/CoreModeMusicController" Controller(
    x::Int,
    y::Int,
    params::String="",
    hot::Number=1.0,
    cold::Number=0.0,
    none::Number=0.0,
    disable::Bool=false
)

const placements = Ahorn.PlacementDict(
    "Core Mode Music Controller (Communal Helper)" => Ahorn.EntityPlacement(
        Controller,
    ),
    "Core Mode Music Controller (Communal Helper) (Disable)" => Ahorn.EntityPlacement(
        Controller,
        "point",
        Dict{String, Any}(
            "disable" => true,
        ),
    ),
)

const spriteEnable = "objects/CommunalHelper/coreModeMusicController/iconEnable"
const spriteDisable = "objects/CommunalHelper/coreModeMusicController/iconDisable"

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::Controller) = Ahorn.drawImage(ctx, Bool(get(entity.data, "disable", false)) ? spriteDisable : spriteEnable, -8, -11)

function Ahorn.selection(entity::Controller)
    x, y = Ahorn.position(entity)
    return Ahorn.Rectangle(x - 8, y - 11, 22, 22)
end

end

File: Loenn/entities/core_mode_music_controller.lua
local coreModeMusicController = {}

coreModeMusicController.name = "CommunalHelper/CoreModeMusicController"
coreModeMusicController.depth = -1000000

function coreModeMusicController.texture(room, entity)
    return entity.disable and "objects/CommunalHelper/coreModeMusicController/iconDisable" or "objects/CommunalHelper/coreModeMusicController/iconEnable"
end

coreModeMusicController.placements = {
    {
        name = "enable",
        data = {
            params = "",
            hot = 1.0,
            cold = 0.0,
            none = 0.0,
            disable = false
        }
    },
    {
        name = "disable",
        data = {
            params = "",
            hot = 1.0,
            cold = 0.0,
            none = 0.0,
            disable = true
        }
    }
}

return coreModeMusicController
